lib/elixir/lib/kernel.ex

```diff
@@ -1570,7 +1570,7 @@ defmodule Kernel do
   """
   @spec inspect(Inspect.t, Keyword.t) :: String.t
   def inspect(arg, opts \\ []) when is_list(opts) do
-    opts  = struct(Inspect.Opts, opts)
+    opts  = struct!(Inspect.Opts, opts)
```

Comment:
backwards incompatible change. inspect will now raise if the options do not always match, we should probably revert this. :)